### DISCUSSION TITLE
New contributor getting an error from make dev

### DISCUSSION THREAD

**Discussion by petr-tik** [ocaml/dune#6758] [General]
Hey, I wanted to start hacking on dune and following instructions here
https://github.com/ocaml/dune/blob/main/doc/hacking.rst#bootstrapping

I thought it would be easy to setup, but got a build error due to a missing dep. 

## Expected Behavior

Running `make dev` as documented in doc/hacking.rst builds successfully

## Actual Behavior

```bash
petr_tik@merluzon:~/Coding/dune main
git show --pretty=oneline | head -1
f27f63e95450b2d8f08053da316c2c881b26b23f Fix for issue #6486 "Inconsistent ppx reporting by dune describe" (#6727)
petr_tik@merluzon:~/Coding/dune main
make dev
ocamlc -output-complete-exe -w -24 -g -o .duneboot.exe -I boot unix.cma boot/libs.ml boot/duneboot.ml
./.duneboot.exe
./_boot/dune.exe build @install
File "otherlibs/dune-rpc-lwt/src/dune", line 4, characters 38-46:
4 |  (libraries result csexp dune_rpc lwt lwt.unix unix)
                                          ^^^^^^^^
Error: Library "lwt.unix" not found.
-> required by library "dune-rpc-lwt" in
   _build/default/otherlibs/dune-rpc-lwt/src
-> required by _build/default/META.dune-rpc-lwt
-> required by _build/install/default/lib/dune-rpc-lwt/META
-> required by _build/default/dune-rpc-lwt.install
-> required by alias install
Makefile:53: recipe for target 'dev' failed          
make: *** [dev] Error 1
```

## Reproduction

1. git clone 
1. checkout the SHA above `git checkout f27f63e95450b2d8f08053da316c2c881b26b23f`
1. Run `make dev`

## Specifications

- Version of `dune`
Not installed 

```bash
dune --version

Command 'dune' not found, but can be installed with:

sudo apt install whitedune
```

- Version of `ocaml` (output of `ocamlc --version`)
```bash
ocamlc --version
4.14.0
petr_tik@merluzon:~/Coding/dune main
which ocamlc
/home/petr_tik/.opam/4.14.0/bin/ocamlc
```

- Operating system (distribution and version):
```bash
lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.5 LTS
Release:	18.04
Codename:	bionic
````

## Additional info

Might be cool to automate build error reporting with a Makefile target that runs `make dev` and then collects everything needed for a bug report (listed above) and helps open an issue.

**Comment by rgrinberg** [comment]
Hey, actually your bootstrap completed successfully. `dune-rpc-lwt` is a package in our repository that depends on lwt, but you don't need to build it to hack on dune unless you're intending to do some very specific stuff.

In the meantime, you can either ignore the error or build a more specific target. For example, `./dune.exe build dune.install`.

**Comment by petr-tik** [comment]
> Hey, actually your bootstrap completed successfully. 

Thanks for clarifying. Getting a non-zero return code confused me and made me think something is wrong. 

Would you be open to changing the `dev` Makefile target to return successfully for new potential contributors to avoid this confusion?

**Reply by rgrinberg** [reply]
I think we can just change the target to build `dune.install` as I've mentioned. Most contributors aren't going to be interested in building dune-rpc-lwt so there's no need for them to see the ugly failure.
